### DISCUSSION TITLE
1.0.18.1 Release Notes

### DISCUSSION THREAD

**Discussion by andrewdewaal** [jedisct1/libsodium#1162] [General]
Hello,

I was unable to find a tag for the 1.0.18.1 release, or any release notes describing what changes were made in this release.

I have gone through the GitHub, the NuGet, and the libsodium project pages, and have been unable to find any content on this.

Can the release notes for 1.0.18.1 be made available, or a tag made in the repo to determine what changes can be expected in this version?

Thanks very much!

**Comment by ektrah** [comment]
1.0.18.1 is a release of the libsodium NuGet package that contains pre-compiled binaries of libsodium 1.0.18 for .NET bindings; it's not a release of libsodium itself. As such, there are no changes to libsodium in this release, only to the NuGet package. There will soon be a 1.0.18.2 release of the libsodium NuGet package, which, again, has no changes to libsodium, but simply adds more pre-compiled binaries for different platforms.

**Reply by andrewdewaal** [reply]
Ah, that makes sense. Thanks very much for the clarification and quick response!

**Reply by samuel-lucas6** [reply]
@ektrah Will that include macOS ARM64? That would be useful for one of my projects that I'm trying to get on Homebrew.

**Reply by ektrah** [reply]
Theoretically yes, practically I’m not sure: Looking at the current packaging files, it seems the GitHub Action [is building](https://github.com/jedisct1/libsodium/blob/2e9b4c47c8cf6759a11a90c2468036e47674b29a/.github/workflows/dotnet-core.yml#L247) a binary for `osx-arm64` but then [not including it](https://github.com/jedisct1/libsodium/blob/2e9b4c47c8cf6759a11a90c2468036e47674b29a/packaging/dotnet-core/libsodium.pkgproj#L35) in the actual NuGet package, as [this line](https://github.com/jedisct1/libsodium/pull/1106/files#diff-f1975c8d358d3e527cb3211e767ebf8d47ade855b09581ca380ee058f91cd4f3R35) somehow went missing between the recent  changes…

/cc @SeanMollet @enclave-alistair @jedisct1

**Reply by SeanMollet** [reply]
@ektrah I'd say that's a correct assessment. Once that line goes back in, it should be golden.

**Reply by ektrah** [reply]
@SeanMollet I wish we had some tests that would catch this 😆

**Reply by SeanMollet** [reply]
@ektrah  I wish Github actions had an apple silicon instance to facilitate running them.  :(
https://github.com/actions/virtual-environments/issues/2187

**Reply by enclave-alistair** [reply]
@ektrah, is there anything remaining to do there to add that osx-arm64 in? Or is it already done?

**Reply by ektrah** [reply]
@enclave-alistair It looks like [libsodium.pkgproj](https://github.com/jedisct1/libsodium/blob/master/packaging/dotnet-core/libsodium.pkgproj) needs to have `<Content Include="runtimes/osx-arm64/native/libsodium.dylib" PackagePath="runtimes/osx-arm64/native/" />` added back in. And it would be great if someone with an M1 Mac could manually test if the .nupkg artifact works on their machine :blush:

**Reply by SeanMollet** [reply]
@ektrah 
PR for the fix, including output of the library tests. I'll have to put together a test of the nupkg. Can't get that done today, but I'll add it to my list hopefully for later this or next week.

https://github.com/jedisct1/libsodium/pull/1164

**Reply by ektrah** [reply]
@SeanMollet Great! Many thanks. For testing the .nupkg artifact, a simple `dotnet new console` with [this Program.cs](https://github.com/jedisct1/libsodium/blob/master/packaging/dotnet-core/test.cs) and `dotnet add package libsodium -s <artifact-dir>` and `dotnet run` should suffice.

**Reply by SeanMollet** [reply]
@ektrah thanks!. Done. I didn't know the -s option for add package, that makes it much easier.
